src/library/iCommand.ts

```diff
@@ -9,5 +10,8 @@ import { Client, Message } from "discord.js";
 export default interface ICommand {
   readonly name: string;
   readonly description: string;
-  execute(args: string[], msg: Message, client?: Client): void;
+  execute(args: string[], msg: Message, extra?: {
+    client?: Client,
+    commands?: Commands
+  }): void;
```

Comment:
Moving towards the optional interface related to #109. 
After these tests get merged we can work on getting the `args` and `msg` into the object as an optional parameter.